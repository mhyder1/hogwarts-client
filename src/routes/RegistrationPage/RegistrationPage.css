.RegistrationPage h2{
    text-shadow: 0 0 3px #F00000;
    font-weight: bold;
    text-decoration: underline;
}