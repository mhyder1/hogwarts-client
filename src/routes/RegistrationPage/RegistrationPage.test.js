import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationPage from './RegistrationPage';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><RegistrationPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <RegistrationPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
