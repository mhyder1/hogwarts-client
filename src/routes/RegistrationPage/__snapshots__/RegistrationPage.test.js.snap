// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<section
  className="Section RegistrationPage"
>
  <h2>
    Register
  </h2>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="preferred_name"
    >
      <label
        htmlFor="RegistrationForm__preferred_name"
      >
        Preferred Name 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="RegistrationForm__preferred_name"
        name="preferred_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="user_name"
    >
      <label
        htmlFor="RegistrationForm__user_name"
      >
        User Name 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="RegistrationForm__user_name"
        name="user_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="RegistrationForm__password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="Button "
      type="submit"
    >
      Register
    </button>
  </form>
</section>
`;
